pkgs/top-level/kodi-packages.nix

```
@@ -34,170 +23,186 @@ let self = rec {
     in
       unique (modules ++ concatLists (catAttrs "requiredKodiAddons" modules));
 
-  # package update scripts
+  self = {
```

The original self was a `rec`, and `kodi` is used in `kodiAddonFor`. Pretty sure this is a semantic change.

```suggestion
  self = rec {
```

---

I don't believe it is...

1. `kodi` comes from line 1 in both diffs.
2. It's bound into the attrset at line 16 left, line 31 right.
3. The same `kodi` is still bound into `hasKodiAddon` and `toKodiAddon`.

Unless there's some subtlety about recursive attrsets that I'm failing to understand, I _think_ it all checks out. Maybe something about the `callPackage` definition and splicing?

I can definitely put the `rec` back: I'm more interested in learning what's missing in my mental model than I am about this particular change being one way or another.

---

> 1. `kodi` comes from line 1 in both diffs.

Ah, I missed that. Seems alright then.